### PR TITLE
feature: generalize modal component

### PR DESCRIPTION
**Pull request by thediaval**

### CODE REVIEW

**Comment by thediaval** [comment]
closes #444

File: npm/ng-packs/packages/identity/src/lib/components/users/users.component.html [hunk 1]
```diff
@@ -77,119 +77,118 @@ <h5 class="card-title">{{ 'AbpIdentity::Users' | abpLocalization }}</h5>
...
-      <ngb-tabset>
-        <ngb-tab [title]="'AbpIdentity::UserInformations' | abpLocalization">
-          <ng-template ngbTabContent>
-            <div class="mt-2">
-              <div class="form-group">
-                <label for="user-name">{{ 'AbpIdentity::UserName' | abpLocalization }}</label
-                ><span> * </span>
-                <input type="text" id="user-name" class="form-control" formControlName="userName" />
-              </div>
+<abp-modal [(visible)]="isModalVisible" *ngIf="isModalVisible" [centered]="true">
```
Comment (mehmet-erim) [inline npm/ng-packs/packages/identity/src/lib/components/users/users.component.html]: Centered input default value is `true`. You can remove `[centered]="true"`

We add structural directives near to HTML tag. Replace below
```html
<abp-modal [(visible)]="isModalVisible" *ngIf="isModalVisible"
```

File: npm/ng-packs/packages/identity/src/lib/components/users/users.component.ts [hunk 2]
```diff
@@ -1,27 +1,37 @@
-import { Component, TemplateRef, ViewChild, TrackByFunction, OnInit } from '@angular/core';
-import { Select, Store } from '@ngxs/store';
-import { Observable, combineLatest, Subject } from 'rxjs';
-import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
-import { FormBuilder, FormGroup, Validators, FormArray, AbstractControl } from '@angular/forms';
-import { IdentityState } from '../../states/identity.state';
-import { Identity } from '../../models/identity';
+import { ABP } from "@abp/ng.core";
```
Comment (mehmet-erim) [inline npm/ng-packs/packages/identity/src/lib/components/users/users.component.ts]: Prettier formatted wrong this file. Please reformat.

File: npm/ng-packs/packages/permission-management/src/lib/components/permission-management.component.html [hunk 3]
```diff
@@ -1,15 +1,9 @@
-<ng-template #modalContent let-modal>
+<abp-modal *ngIf="visible" size="lg" [(visible)]="visible" [centered]="true">
```
Comment (mehmet-erim) [inline npm/ng-packs/packages/permission-management/src/lib/components/permission-management.component.html]: Size and centered are unnecessary.

File: npm/ng-packs/packages/tenant-management/src/lib/components/tenants/tenants.component.html [hunk 4]
```diff
@@ -88,60 +88,62 @@ <h5 class="card-title">
...
-<ng-template #modalWrapper let-modal>
-  <div class="modal-header">
-    <h5 class="modal-title" id="modal-basic-title">
-      {{ selectedModalContent.title | abpLocalization }}
-    </h5>
-    <button type="button" class="close" aria-label="Close" (click)="modal.dismiss()">
-      <span aria-hidden="true">&times;</span>
-    </button>
-  </div>
+<abp-modal *ngIf="isModalVisible" size="md" [(visible)]="isModalVisible" [centered]="true">
```
Comment (mehmet-erim) [inline npm/ng-packs/packages/tenant-management/src/lib/components/tenants/tenants.component.html]: Please remove size and centered

File: npm/ng-packs/packages/tenant-management/src/lib/components/tenants/tenants.component.html [hunk 5]
```diff
@@ -88,60 +88,62 @@ <h5 class="card-title">
...
 </ng-template>
 
-<ng-template #mTemplateConnStr let-modal>
-  <div class="mt-2">
-    <div class="form-group">
-      <div class="form-check">
-        <input id="useSharedDatabase" type="checkbox" class="form-check-input" formControlName="useSharedDatabase" />
-        <label for="useSharedDatabase" class="font-check-label">{{
-          'AbpTenantManagement::DisplayName:UseSharedDatabase' | abpLocalization
+<ng-template #ConnectionStringModalTemplate>
```
Comment (mehmet-erim) [inline npm/ng-packs/packages/tenant-management/src/lib/components/tenants/tenants.component.html]: First letter must be lower

File: npm/ng-packs/packages/tenant-management/src/lib/components/tenants/tenants.component.html [hunk 6]
```diff
@@ -88,60 +88,62 @@ <h5 class="card-title">
...
+    </button>
+  </ng-template>
+</abp-modal>
 
-<ng-template #mTemplateTenant let-modal>
-  <div class="mt-2">
-    <div class="form-group">
-      <label for="name">{{ 'AbpTenantManagement::TenantName' | abpLocalization }}</label>
-      <input type="text" id="name" class="form-control" formControlName="name" />
+<ng-template #TenantModalTemplate>
```
Comment (mehmet-erim) [inline npm/ng-packs/packages/tenant-management/src/lib/components/tenants/tenants.component.html]: First letter must be lower

File: npm/ng-packs/packages/tenant-management/src/lib/components/tenants/tenants.component.html [hunk 7]
```diff
@@ -88,60 +88,62 @@ <h5 class="card-title">
...
-        'AbpTenantManagement::DisplayName:DefaultConnectionString' | abpLocalization
-      }}</label>
-      <input type="text" id="defaultConnectionString" class="form-control" formControlName="defaultConnectionString" />
-    </div>
-  </div>
+  </form>
 </ng-template>
 
-<ng-template #mTemplateFeatures let-modal>
+<ng-template #FeaturesModalTemplate>
```
Comment (mehmet-erim) [inline npm/ng-packs/packages/tenant-management/src/lib/components/tenants/tenants.component.html]: First letter :)

File: npm/ng-packs/packages/tenant-management/src/lib/components/tenants/tenants.component.ts [hunk 8]
```diff
@@ -31,46 +36,36 @@ export class TenantsComponent {
 
   defaultConnectionString: string;
 
-  useSharedDatabase: boolean;
+  _useSharedDatabase: boolean;
```
Comment (mehmet-erim) [inline npm/ng-packs/packages/tenant-management/src/lib/components/tenants/tenants.component.ts]: is the first letter necessary? `_`

Comment (mehmet-erim) [reply npm/ng-packs/packages/tenant-management/src/lib/components/tenants/tenants.component.ts]: It's necessary, I saw it. But if you are move this variable the 44th line, reading will be easier.

File: npm/ng-packs/packages/tenant-management/src/lib/components/tenants/tenants.component.ts [hunk 9]
```diff
@@ -100,56 +100,59 @@ export class TenantsComponent {
...
+      this.tenantService
+        .deleteDefaultConnectionString(this.selected.id)
+        .pipe(take(1))
+        .subscribe(() => {
+          this.isModalVisible = false;
+        });
+    } else {
+      this.tenantService
+        .updateDefaultConnectionString({ id: this.selected.id, defaultConnectionString: this.connectionString })
+        .pipe(take(1))
```
Comment (mehmet-erim) [inline npm/ng-packs/packages/tenant-management/src/lib/components/tenants/tenants.component.ts]: You added `take(1)`. Did you check the subscription status?

File: npm/ng-packs/packages/theme-basic/src/lib/components/profile/profile.component.html [hunk 10]
```diff
@@ -26,22 +23,21 @@ <h4 class="modal-title" id="modal-basic-title">{{ 'AbpIdentity::PersonalInfo' |
...
-      <button type="submit" class="btn btn-primary">
-        <i class="fa fa-check mr-1"></i> <span>{{ 'AbpIdentity::Save' | abpLocalization }}</span>
-      </button>
-    </div>
-  </form>
-</ng-template>
+    </form>
+  </ng-template>
+  <ng-template #abpFooter>
+    <button type="button" class="btn btn-secondary" #abpClose>
```
Comment (mehmet-erim) [inline npm/ng-packs/packages/theme-basic/src/lib/components/profile/profile.component.html]:
```
    <button #abpClose type="button" class="btn btn-secondary">
```

**Pull request merged**